build.sbt

```diff
@@ -1,120 +1,110 @@
-  "org.http4s"                   %% "http4s-blaze-client"           % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-dsl"                    % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-circe"                  % http4sVersion % IntegrationTest,
-  compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
-  "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
-)
+lazy val conseil = (project in file("."))
+  .disableAssembly()
+  .aggregate(common, api, lorre, lorreSchema, smokeTests)
 
-excludeDependencies ++= Seq(
-  "org.consensusresearch" %% "scrypto"
-)
+lazy val common = (project in file("conseil-common"))
+  .settings(
+    name := "conseil-common",
+    libraryDependencies ++= Dependencies.conseilCommonInclude,
+    excludeDependencies ++= Dependencies.conseilCommonExclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          "tech.cryptonomic.conseil.common.io.*",
+          "tech.cryptonomic.conseil.common.tezos.Tables",
+          "tech.cryptonomic.conseil.common.config.Security"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .settings(
+    scalacOptions += "-P:silencer:pathFilters=common/src/main/scala/tech/cryptonomic/conseil/common/tezos/Tables.scala"
+  )
+  .configs(IntegrationTest)
+  .enableBuildInfo()
+  .disableAssembly()
 
-assemblyOutputPath in assembly := file("/tmp/conseil.jar")
+lazy val api = (project in file("conseil-api"))
+  .settings(
+    name := "conseil-api",
+    mainClass := Some("tech.cryptonomic.conseil.api.Conseil"),
+    libraryDependencies ++= Dependencies.conseilApiInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.Conseil",
+          ".*\\.ConseilAppConfig"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .configs(IntegrationTest)
+  .addRunCommand(
+    description = "A run application Task.",
+    javaExtras = Seq("-Xms1024M", "-Xmx8192M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
+  )
+  .enableAssembly()
+  .dependsOn(common % "compile->test")
 
-scalacOptions ++= ScalacOptions.common
-scalacOptions += "-P:silencer:pathFilters=src/main/scala/tech/cryptonomic/conseil/tezos/Tables.scala"
-import complete.DefaultParsers._
+lazy val lorre = (project in file("conseil-lorre"))
+  .settings(
+    name := "conseil-lorre",
+    mainClass := Some("tech.cryptonomic.conseil.lorre.Lorre"),
+    libraryDependencies ++= Dependencies.conseilLorreInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.Lorre",
+          ".*\\.LorreAppConfig"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .configs(IntegrationTest)
+  .addRunCommand(
+    description = "A run application Task.",
+    javaExtras = Seq("-Xmx512M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
```

Comment:
I think Lorre needs bigger Xmx. for development I use 8192M

Reply:
Well, I copied that from original code, so I don't know why it should be changed now?
This change may have bigger impact that we think, so I would rather suggest to change that later, in separate PR.

Old code for reference: https://github.com/Cryptonomic/Conseil/blob/master/build.sbt#L113

Reply:
I guess we're overriding those values locally anyway.

@piotrkosecki are we sure about the relevant values needed here?
In any case we might adapt them after this PR if needed.

Reply:
I remember when I tried running Lorre for longer time to get some data for testing I needed to use bigger Xmx because it crashed relatively often. TBH I'm not sure how it works right now, as I'm always using those values.

Reply:
We agreed that we are going to change that if needed in upcoming pull-requests.